// @magobaol/phil/lib/Filters/caseConverters.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.toSpinal = exports.toSnake = exports.toPascal = exports.toCamel = exports.toSentence = exports.toTitle = exports.toLower = exports.toUpper = void 0;
const respacer_1 = require("./respacer");
const flatten = (text) => {
    return (0, exports.toLower)((0, respacer_1.toSpace)(text));
};
//UPPER CASE
const toUpper = (text) => {
    return text.toUpperCase();
};
exports.toUpper = toUpper;
exports.toUpper.filterName = 'toUpper';
//lower case
const toLower = (text) => {
    return text.toLowerCase();
};
exports.toLower = toLower;
exports.toLower.filterName = 'toLower';
//Title Case
const toTitle = (text) => {
    const flat = flatten(text);
    return flat.replace(/\b\w/g, (match) => match.toUpperCase());
};
exports.toTitle = toTitle;
exports.toTitle.filterName = 'toTitle';
//Sentence case
const toSentence = (text) => {
    const flat = flatten(text);
    return flat.replace(/^[a-z]{1}/g, (match) => match.toUpperCase());
};
exports.toSentence = toSentence;
exports.toSentence.filterName = 'toSentence';
//camelCase
const toCamel = (text) => {
    const flat = flatten(text);
    return flat
        .replace(/^\w|[A-Z]|\b\w/g, (word, index) => (index === 0 ? word.toLowerCase() : word.toUpperCase()))
        .replace(/\s+/g, '');
};
exports.toCamel = toCamel;
exports.toCamel.filterName = 'toCamel';
//PascalCase
const toPascal = (text) => {
    const flat = flatten(text);
    return flat.replace(/^\w|[A-Z]|\b\w/g, (word) => word.toUpperCase()).replace(/\s+/g, '');
};
exports.toPascal = toPascal;
exports.toPascal.filterName = 'toPascal';
//snake_case
const toSnake = (text) => {
    const flat = flatten(text);
    return flat
        .replace(/[^\w\s]/gi, '')
        .replace(/\s+/g, '_')
        .toLowerCase();
};
exports.toSnake = toSnake;
exports.toSnake.filterName = 'toSnake';
//spinal-case
const toSpinal = (text) => {
    const flat = flatten(text);
    return flat
        .replace(/[^\w\s]/gi, '')
        .replace(/\s+/g, '-')
        .toLowerCase();
};
exports.toSpinal = toSpinal;
exports.toSpinal.filterName = 'toSpinal';
